Support MultiverseBridge (AKA CardSphere) IDs & Prices

ZeldaZach opened the pull request:
Also abstract away part of how prices are generated for providers, since it's a lot of duplicated code.
(Will work more towards converting other price builders to use the generic method if possible.)

ADD card.identifiers.cardsphere_id -- Optional String
ADD set.cardsphere_set_id -- Optional Integer
ADD CardSphere to prices file under paper category with USD -- Retail only, No Buylist

noob247365 commented on mtgjson5/price_builder.py:
```diff
@@ -74,9 +75,12 @@ def build_today_prices() -> Dict[str, Any]:
     tcgplayer = _generate_prices(TCGPlayerProvider())
     card_market = _generate_prices(CardMarketProvider())
     card_kingdom = _generate_prices(CardKingdomProvider())
+    cardsphere = _generate_prices(MultiverseBridgeProvider())
```
Naming is not consistent. Should either rename provider class, or rename variable to match

noob247365 replied:
After further review, would side with renaming the variable over renaming the class

ZeldaZach replied:
The naming is intentionally different, as MB can/will provide other price Ids in the future, and this sets a precedent that we can pull it in multiple ways. If my logic here is wrong, I'm fine with changing it

noob247365 replied:
Oh, makes enough sense. Then yeah, keeping it is fine 👍

ZeldaZach committed:
Add logging details for price building on other entities

ZeldaZach committed:
Abstract away how CardKingdom prices are generated for future use

ZeldaZach committed:
Support MultiverseBridge (AKA CardSphere) IDs & Prices

Also abstract away part of how prices are generated for providers, since it's a lot of duplicated code.
(Will work more towards converting other price builders to use the generic method if possible.)

ADD card.identifiers.cardsphere_id -- Optional String
ADD set.cardsphere_set_id -- Optional Integer
ADD CardSphere to prices file under paper category with USD -- Retail only, No Buylist

noob247365 reviewed (changes requested):
Just one variable name that's inconsistent, rest seems fine (for as much as I understand about this project :P)

kodawah commented on mtgjson5/classes/mtgjson_identifiers.py:
```diff
@@ -14,9 +14,11 @@ class MtgjsonIdentifiersObject:
     mcm_id: Optional[str]
     mcm_meta_id: Optional[str]
     mtg_arena_id: Optional[str]
+    mtgjson_v4_id: Optional[str]
```
unneeded move
